constraints/miden-vm/stack.air

```diff
@@ -0,0 +1,1015 @@
+
+
+# High-degree operations
+
+fn f_hperm(op_bits: vector[8]) -> scalar:
+    return f_101(op_bits) & !op_bits[3] & !op_bits[2] & !op_bits[1]
+
+fn f_mpverify(op_bits: vector[8]) -> scalar:
+    return f_101(op_bits) & !op_bits[3] & !op_bits[2] & op_bits[1]
+
+fn f_pipe(op_bits: vector[8]) -> scalar:
+    return f_101(op_bits) & !op_bits[3] & op_bits[2] & !op_bits[1]
+
+fn f_mstream(op_bits: vector[8]) -> scalar:
+    return f_101(op_bits) & !op_bits[3] & op_bits[2] & op_bits[1]
+
+fn f_span(op_bits: vector[8]) -> scalar:
+    return f_101(op_bits) & op_bits[3] & !op_bits[2] & !op_bits[1]
+
+fn f_join(op_bits: vector[8]) -> scalar:
+    return f_101(op_bits) & op_bits[3] & !op_bits[2] & op_bits[1]
+
+fn f_split(op_bits: vector[8]) -> scalar:
+    return f_101(op_bits) & op_bits[3] & op_bits[2] & !op_bits[1]
+
+fn f_loop(op_bits: vector[8]) -> scalar:
+    return f_101(op_bits) & op_bits[3] & op_bits[2] & op_bits[1]
+
+
+# Very high-degree operations
+
+fn f_mrupdate(op_bits: vector[8]) -> scalar:
+    return op_bits[7] & !op_bits[4] & !op_bits[3] & !op_bits[2]
+
+fn f_push(op_bits: vector[8]) -> scalar:
+    return op_bits[7] & !op_bits[4] & !op_bits[3] & op_bits[2]
+
+fn f_syscall(op_bits: vector[8]) -> scalar:
+    return op_bits[7] & !op_bits[4] & op_bits[3] & !op_bits[2]
+
+fn f_call(op_bits: vector[8]) -> scalar:
+    return op_bits[7] & !op_bits[4] & op_bits[3] & op_bits[2]
+
+fn f_end(op_bits: vector[8]) -> scalar:
+    return op_bits[7] & op_bits[4] & !op_bits[3] & !op_bits[2]
+
+fn f_repeat(op_bits: vector[8]) -> scalar:
+    return op_bits[7] & op_bits[4] & !op_bits[3] & op_bits[2]
+
+fn f_respan(op_bits: vector[8]) -> scalar:
+    return op_bits[7] & op_bits[4] & op_bits[3] & !op_bits[2]
+
+fn f_halt(op_bits: vector[8]) -> scalar:
+    return op_bits[7] & op_bits[4] & op_bits[3] & op_bits[2]
+
+
+# Composite flags
+
+fn f_shr(op_bits: vector[8]) -> scalar:
+    return !op_bits[6] & op_bits[5] & op_bits[4] + f_u32split(op_bits) + f_push(op_bits)
+
+# hasher[5] = op_helpers[3], where hahser -- main columns from the decoder, helper -- aux columns from the stack
+fn f_shl(op_bits: vector[8], op_helpers: vector[6]) -> scalar:
+    let f_add3_mad = op_bits[6] & !op_bits[5] & !op_bits[4] & op_bits[3] & op_bits[2]
+    let f_split_loop = op_bits[6] & !op_bits[5] & op_bits[4] & op_bits[3] & op_bits[2]
+    return !op_bits[6] & op_bits[5] & !op_bits[4] + f_add3_mad + f_split_loop + f_repeat(op_bits) + f_end(op_bits) * op_helpers[3]
+
+fn f_ctrl(op_bits: vector[8]) -> scalar:
+    # flag for SPAN, JOIN, SPLIT, LOOP
+    let f_sjsl = op_bits[6] & !op_bits[5] & op_bits[4] & op_bits[3]
+
+    # flag for END, REPEAT, RESPAN, HALT
+    let f_errh = op_bits[6] & op_bits[5] & op_bits[4]
+
+    return f_sjsl + f_errh + f_call(op_bits) + f_syscall(op_bits)
+
+
+fn compute_op_flags(op_bits: vector[8]) -> vector[88]:
+    return [
+        f_noop(op_bits),
+        f_eqz(op_bits),
+        f_neg(op_bits),
```

Comment:
I'm not sure putting each operation flag into a separate function gives us much in terms of modularity or readability. Maybe we should get rid of individual functions and just add some comments here? For example, this could look like:
```
fn compute_op_flags(op_bits: vector[8]) -> vector[88]:
    return [
        f_000(op_bits) & f_x0000(op_bits), # noop
        f_000(op_bits) & f_x0001(op_bits), # eqz
        f_000(op_bits) & f_x0010(op_bits), # neg
```

Reply:
I agree. I thought that we would use each flag individually, but since we have the `compute_op_flags` function they lose their main purpose.